website/project/mailing_list.py

```diff
@@ -0,0 +1,140 @@
+from website.models import Node, NotificationSubscription
+from website.notifications.utils import to_subscription_key
+
+from website.project.model import MailingListEventLog
+from website.project.signals import contributor_added
+from website.util.sanitize import unescape_entities
+
+###############################################################################
+# Signalled Functions
+###############################################################################
+
+@contributor_added.connect
+def subscribe_contributor_to_mailing_list(node, contributor, auth=None):
+    if node.mailing_enabled and contributor.is_active:
+        subscription = NotificationSubscription.load(
+            to_subscription_key(node._id, 'mailing_list_events')
+        )
+        subscription.add_user_to_subscription(contributor, 'email_transactional', save=True)
+        subscription.save()
+
+@user_confirmed.connect
+def resubscribe_on_confirm(user):
+    for node in user.contributed:
+        subscribe_contributor_to_mailing_list(node, user)
+
+###############################################################################
+# Mailing List Functions
+###############################################################################
+
+def address(node_id):
+    return node_id + '@' + settings.SHORT_DOMAIN
+
+def find_email(long_email):
+    # allow for both "{email}" syntax and "{name} <{email}>" syntax
+    if ' ' in long_email:
+        email = re.search(r'<\S*>$', long_email).group(0)[1:-1]
+        return email
+    else:
+        return long_email
+
+
+def route_message(**kwargs):
+    """ Acquires messages sent through Mailgun, validates them, and warns the
+    user if they are not valid"""
+    message = request.form
+    target = find_email(message['To'])
+    node_id = re.search(r'[a-z0-9]*@', target).group(0)[:-1]
+    node = Node.load(node_id)
+
+    sender_email = find_email(message['From'])
+    sender = get_user(email=sender_email)
+
+    user_is_admin = 'admin' in node.get_permissions(sender)\
+        if sender and node else False
+
+    mail_params = {
+        'to_addr': sender_email,
+        'mail': mails.DISCUSSIONS_EMAIL_REJECTED,
+        'target_address': target,
+        'user': sender,
+        'node_type': node.project_or_component if node else '',
+        'node_url': node.absolute_url if node else '',
+        'is_admin': user_is_admin,
+        'mail_log_class': MailingListEventLog
+    }
+
+    if not sender:
+        reason = MailingListEventLog.UNAUTHORIZED
+    elif not node:
+        reason = MailingListEventLog.NOT_FOUND
+    elif node.is_deleted:
+        reason = MailingListEventLog.DELETED
+    elif sender not in node.contributors:
+        reason = MailingListEventLog.FORBIDDEN
+    elif not node.mailing_enabled:
+        reason = MailingListEventLog.DISABLED
+    else:
+        reason = ''
+
+    if reason:
+        mails.send_mail(reason=reason, **mail_params)
+    else:
+        send_messages(node, sender, message)
```

Comment:
... rename `send_messages` to be parallel to `send_rejection`, I guess `send_accepted` would work.